src/Ztm.Zcoin.Rpc.Tests/ZcoinRpcClientTests.cs

```diff
@@ -456,5 +456,226 @@ public async Task SendToAddressAsync_WithNullAmount_ShouldThrow()
+            Assert.Equal(new PropertyAmount(10), destinationBalance.Balance);
+        }
+
+        [Fact]
+        public async Task GetPropertyBalanceAsync_WithNullArgs_ShouldThrow()
+        {
+            // Arrange.
+            var builder = new PropertyBuilder(this.node, this.subject);
+            var managedProperty = await builder.BuildAsync(CancellationToken.None);
+            var address = await this.subject.GetNewAddressAsync(CancellationToken.None);
+
+            // Assert.
+            await Assert.ThrowsAsync<ArgumentNullException>(
+                "address",
+                () => this.subject.GetPropertyBalanceAsync(null, managedProperty.property, CancellationToken.None)
+            );
+
+            await Assert.ThrowsAsync<ArgumentNullException>(
+                "property",
+                () => this.subject.GetPropertyBalanceAsync(address, null, CancellationToken.None)
+            );
+        }
+
+        [Fact]
+        public async Task GetPropertyBalanceAsync_WithValidArgs_ShouldSuccess()
+        {
+            // Arrange.
+            var builder = new PropertyBuilder(this.node, this.subject);
+            builder.propertyType = PropertyType.Indivisible;
+            var property = await builder.BuildAsync(CancellationToken.None);
+
+            await property.GrantAsync(1000);
+
+            // Act.
+            var balance = await this.subject.GetPropertyBalanceAsync(
+                property.owner, property.property, CancellationToken.None);
+
+            // Assert.
+            Assert.Equal(new PropertyAmount(1000), balance.Balance);
+            Assert.Equal(new PropertyAmount(0), balance.Reserved);
+        }
+
+        [Fact]
+        public async Task GetPropertyBalanceAsync_WithDivisibleToken_ShouldSuccess()
+        {
+            // Arrange.
+            var builder = new PropertyBuilder(this.node, this.subject);
+            builder.propertyType = PropertyType.Divisible;
+            var property = await builder.BuildAsync(CancellationToken.None);
+
+            await property.GrantAsync(10_0000_0000);
+
+            // Act.
+            var balance = await this.subject.GetPropertyBalanceAsync(
+                property.owner, property.property, CancellationToken.None);
+
+            // Assert.
+            Assert.Equal(PropertyAmount.FromDivisible(10), balance.Balance);
+            Assert.Equal(PropertyAmount.FromDivisible(0), balance.Reserved);
+        }
+
+        class ManagedProperty
+        {
+            readonly CoreNode node;
+            readonly ZcoinRpcClient client;
+
+            public BitcoinAddress owner { get; set; }
+            public Property property { get; set; }
+
+            public ManagedProperty(CoreNode node, ZcoinRpcClient client)
+            {
+                this.node = node;
+                this.client = client;
+            }
+
+            public async Task GrantAsync(long grantingAmount)
```

Comment:
Why we not accept `PropertyAmount` instead of `long`?

Reply:
It is good idea to change it to PropertyAmount to support both divisible and indivisible